astropy/wcs/wcs.py

```diff
@@ -438,13 +439,15 @@ def __init__(
             sip = None
         else:
             keysel_flags = _parse_keysel(keysel)
-
-            if isinstance(header, (str, bytes)):
+            if isinstance(header, (str, bytes, os.PathLike)):
                 try:
-                    is_path = os.path.exists(header)
+                    is_path = os.path.exists(header)  # noqa: PTH110
```

Comment:
Can one not replace with `Path(header).exists()`?

Reply:
that would be a comment for #16936, but yes, we could avoid a noqa here